### ISSUE TITLE
[Guidelines] Add guideline on Authentication and Authorization

### ISSUE THREAD

**Issue by yanniszark** [kubeflow/community#327]
Add a Guideline on how to perform Authorization for Kubeflow's Web Applications.
Authorization should rely on K8s RBAC as the source of truth.
For that reason, focus on SubjectAccessReview.

**Comment by jlewi** [comment]
@yanniszark Are you planning on writing up a doc covering the points discussed in the thread?
https://groups.google.com/forum/#!searchin/kubeflow-discuss/subject$20access$20review%7Csort:date/kubeflow-discuss/whS_j4RlEJo/98xnBHH4BgAJ

In particular

* Getting rid of asymmetric permission model
* Making profiles a namespace scoped resource
* Possibly in scope - how we make namespace setup more extensible and not make profile controller this monolithic thing

**Comment by yanniszark** [comment]
@jlewi yes, this is a doc I want to write, but it's so general I find it a bit hard to think of a proper issue title.
I think this issue should remain seperate and we should open a new one for the doc you mention.
Tbh, I haven't thought about making Profiles namespaced (was this mentioned somewhere?), but the discussion in the aforementioned thread is what I want to capture.
Something like `Evaluate and Improve our current Multi-User story`.
I will open a new issue for that, let's leave this one for the guideline on authorization (use SubjectAccessReview in order to depend on K8s RBAC).

**Comment by jlewi** [comment]
@yanniszark We can use this issue to track the doc. I just wanted to clarify what the deliverable is.
